### PR TITLE
Support permission filter for roles

### PR DESCRIPTION
**Pull request by astrozzc**
## Link(s) to Jira
- JIRA: https://issues.redhat.com/browse/RHCLOUD-9210

## Description of Intent of Change(s)
Add a filter to support filter roles by permission, so that
using ?permission=xxx:xxx:xxx,xxx2:xxx2:xxx2 would return roles
that contain these permissions in their access object.

## Local Testing
Unit tests

## Checklist
- [x] if API spec changes are required, is the spec updated?

- [ ] are there any pre/post merge actions required? if so, document here.

- [ ] are theses changes covered by unit tests?

- [ ] if warranted, are documentation changes accounted for?

- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?

- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation

- [ ] Output Encoding

- [ ] Authentication and Password Management

- [ ] Session Management

- [ ] Access Control

- [ ] Cryptographic Practices

- [ ] Error Handling and Logging

- [ ] Data Protection

- [ ] Communication Security

- [ ] System Configuration

- [ ] Database Security

- [ ] File Management

- [ ] Memory Management

- [ ] General Coding Practices

### CODE REVIEW

**Review by wcmitchell — APPROVED** [review]
Works for me in local testing. 👍

File: rbac/management/role/view.py [hunk 1]
```diff
@@ -70,8 +70,15 @@ def application_filter(self, queryset, field, values):
             query = query | Q(access__permission__istartswith=f"{application}:")
         return queryset.distinct().filter(query)
 
+    def permission_filter(self, queryset, field, values):
+        """Filter to lookup role by application(s) in permissions."""
+        permissions = values.split(",")
+
+        return queryset.filter(access__perm__in=permissions).distinct()
```
Comment (coderbydesign) [inline rbac/management/role/view.py:77]: So this is what we'd update when the refactor to use `Permission` happens?

**Review by coderbydesign — APPROVED** [review]
Had one question, but looks good! 🌴

File: rbac/management/role/view.py (hunk 1 above)
Comment (astrozzc) [reply rbac/management/role/view.py:77]: Right, after the permission association, we only have to refactor here, which is simple 😁

**Pull request merged**